src/app/src/main/kotlin/org/vaccineimpact/api/app/security/JooqOneTimeTokenChecker.kt

```diff
@@ -0,0 +1,15 @@
+package org.vaccineimpact.api.app.security
+
+import org.vaccineimpact.api.app.repositories.RepositoryFactory
+import org.vaccineimpact.api.security.OneTimeTokenChecker
+
+class JooqOneTimeTokenChecker(private val repositoryFactory: RepositoryFactory) : OneTimeTokenChecker
+{
+    override fun checkToken(token: String): Boolean
+    {
+        // This transaction is immediately committed, regardless of result
+        return repositoryFactory.inTransaction { repos ->
+            repos.token.validateOneTimeToken(token)
+        }
+    }
+}
```

Comment:
By wrapping this logic in a simple interface (`OneTimeTokenChecker`) it allows the code to in the security module to remain blissfully unaware of Jooq.

Reply:
:+1: